lib/ansible/plugins/inventory/vmware_inventory.py

```diff
@@ -0,0 +1,410 @@
+#
+# Copyright: (c) 2018, Ansible Project
+# Copyright: (c) 2018, Abhijeet Kasurde <akasurde@redhat.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: vmware_inventory
+    plugin_type: inventory
+    short_description: VMware inventory source
+    description:
+        - Get inventory hosts from VMware environment.
+        - Uses any file which ends with vmware.yml or vmware.yaml as a YAML configuration file.
+        - 'Please make sure you mention "enable_plugins = vmware_inventory" in inventory section of ansible.cfg.'
+        - The inventory_hostname is always the 'Name' and UUID of the virtual machine. UUID is added as VMware allows virtual machines with the same name.
```

Comment:
you might want to make this configurable, some environments force unique names

Reply:
For first draft, I matched the existing inventory script. But, definitely going forward we can have this as a configurable parameter. Thanks for the suggestion.